### DISCUSSION TITLE
Passwordless authentication

### DISCUSSION THREAD

**Discussion by nicolas-besnard** [jeremyevans/rodauth#287] [General]
I'm trying to use Rodauth for passwordless authentication. When the user created his account and came back to the website to confirm his email, an empty page with only a "verify account" button appeared.

Is there a way to validate the account directly, without showing this page and redirecting the user to a configured page?

**Comment by janko** [comment]
You can autosubmit the form on render, as described in [this wiki page](https://github.com/janko/rodauth-rails/wiki/Autosubmit-Forms).

**Reply by nicolas-besnard** [reply]
This is one solution, but the UX is not ideal. Is there a way to have a hook when the page load, just like we have for `before_verify_account` for example?

**Reply by janko** [reply]
In what way is the UX not ideal? Note that the guide suggests hiding the form on render. But I suppose it might look strange depending on the page layout, if that's what you meant.

Hooking on page load is a frontend thing, so Rodauth won't expose anything for that, since that's on the backend. Maybe there are alternatives I'm not seeing.

**Reply by nicolas-besnard** [reply]
Yes, speaking about the page layout + having a blank page (or even a page with style) that appears and then you are directed somewhere. I don't like to rely on javascript to expect a simple behaviour like that.

I could design a specific page with a text like "activating your account...' though 🤔 

It can be a hook or a different logic in the verify_accout module. By default, the page displays a form (mostly for password), but if you have nothing to input, you can have an option to directly validate the account and redirect the user like a `auto_verify_if_no_required_field true`

**Reply by jeremyevans** [reply]
By design, Rodauth does not use GET requests that change state, as GET requests should be idempotent.  The form on the displayed page submits a POST request to change state.  I don't think it makes sense to add a configuration option to make it easier to allow a GET request to change state.
